beamer/deploy/commands.py

```diff
@@ -88,3 +88,96 @@ def deploy_base(
+    required=True,
+    metavar="PATH",
+    help="Path to the keystore file.",
+)
+@click.password_option(
+    "--password",
+    type=str,
+    default="",
+    prompt=False,
+    help="The password needed to unlock the keystore file.",
+)
+@click.option(
+    "--artifacts-dir",
+    type=click.Path(exists=True, file_okay=False, dir_okay=True, path_type=Path),
+    required=True,
+    metavar="DIR",
+    help="The directory to store contract deployment artifacts in.",
+)
+@click.option(
+    "--deploy-mintable-token",
+    is_flag=True,
+    help="Deploy MintableToken.sol on the specified chains.",
+)
+@click.argument(
+    "chain.json",
+    nargs=-1,
+    required=True,
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+)
+def deploy(
+    rpc_file: Path,
+    keystore_file: Path,
+    password: str,
+    artifacts_dir: Path,
+    deploy_mintable_token: bool,
+    **chains: dict,
+) -> None:
+    """Deploy L2 Beamer contracts on the specified chains and set up the trusted call chain."""
+    beamer.util.setup_logging(log_level="DEBUG", log_json=False)
+    _ensure_commit_is_on_remote()
+
+    rpc_info = beamer.deploy.config.load_rpc_info(rpc_file)
+    log.info("Loaded RPC file")
+
+    account = beamer.util.account_from_keyfile(keystore_file, password)
+    log.info("Loaded keystore file", address=account.address)
+
+    path = artifacts_dir / "base.deployment.json"
+    base_deployment = beamer.deploy.artifacts.Deployment.from_file(path)
+    log.info(
+        "Loaded base chain deployment", path=str(path), chain_id=base_deployment.base.chain_id
+    )
+
+    url = rpc_info[base_deployment.base.chain_id]
+    base_w3 = make_web3(url, account)
+    assert base_w3.eth.chain_id == base_deployment.base.chain_id
+    log.info("Connected to base chain RPC", chain_id=base_deployment.base.chain_id, url=url)
+
+    resolver = base_deployment.obtain_contract(base_w3, "base", "Resolver")
+
+    for path in chains["chain.json"]:
+        chain = beamer.deploy.config.Chain.from_file(path)
+        log.info("Loaded chain config", name=chain.name, chain_id=chain.chain_id, path=str(path))
+
+        url = rpc_info[chain.chain_id]
+        w3 = make_web3(url, account)
+        assert w3.eth.chain_id == chain.chain_id
+        log.info("Connected to chain RPC", chain_id=chain.chain_id, url=url)
+
+        l1_contracts, l2_contracts = deploy_beamer(base_w3, chain, resolver)
+        if deploy_mintable_token:
+            l2_contracts += (deploy_contract(w3, ("MintableToken", int(1e18))),)
+
+        path = artifacts_dir / f"{chain.chain_id}-{chain.name}.deployment.json"
+        beamer.deploy.artifacts.generate(path, account.address, l1_contracts, l2_contracts)
+        log.info("Generated artifact", path=str(path))
+
+        path = artifacts_dir / "base.deployment.json"
+        base_deployment.to_file(path)
+        log.info("Updated base chain deployment", path=str(path))
```

Comment:
This is not needed anymore.

Reply:
Gah, don't know how I overlooked that. Removed, thanks!